# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1272666501 14400
#      Fri Apr 30 18:28:21 2010 -0400
# Node ID 4b421123fd175ea1daaed9b74e5c17574d7cd908
# Parent  26d51c6e1c4c0a78fc5ae490acc1b03d4bfa7123
__errplot__.m: Fix bug parsing errorbar style.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2010-04-30  Ben Abbott <bpabbott@mac.com>
 
+	* plot/private/__errplot__.m: Fix bug parsing errorbar style.
 	* plot/private/__errplot__.m: Implement errorbars in the Matlab style.
 
 2010-04-29  Ben Abbott <bpabbott@mac.com>
 
 	* plot/errorbar.m: Modify demo.
 	* plot/private/__errplot__.m: Fix bug for boxxy errorbar.
 	* plot/errorbar.m: Add demos.
 
diff --git a/scripts/plot/private/__errplot__.m b/scripts/plot/private/__errplot__.m
--- a/scripts/plot/private/__errplot__.m
+++ b/scripts/plot/private/__errplot__.m
@@ -71,54 +71,54 @@ function h = __errplot__ (fstr, p, varar
                    "color", fmt.color)),
           (__line__ (hg, "linestyle", "-", "marker", "none",
                    "color", fmt.color))];
 
     switch (numel(varargin))
       case 2
         ydata = varargin{1}(:,i);
         xdata = 1:numel(ydata);
-        if (strcmp (ifmt, "xerr") || index (ifmt, "box"))
+        if (strcmp (ifmt, "xerr") || strcmp (ifmt, "box"))
           xldata = varargin{2}(:,i);
           xudata = ldata;
           ldata = [];
           udata = [];
-        elseif (strcmp (ifmt, "yerr") || index (ifmt, "boxy"))
+        elseif (strcmp (ifmt, "yerr") || strcmp (ifmt, "boxy"))
           ldata = varargin{2}(:,i);
           udata = ldata;
           xldata = [];
           xudata = [];
         else
           error ("2 column errorplot is only valid or xerr or yerr");
         endif
       case 3
-        if (strcmp (ifmt, "boxxy") || index (ifmt, "xyerr"))
+        if (strcmp (ifmt, "boxxy") || strcmp (ifmt, "xyerr"))
           ydata = varargin{1}(:,i);
           xdata = 1:numel(ydata);
           xldata = varargin{2}(:,i);
           xudata = xldata;
           ldata = varargin{3}(:,i);
           udata = ldata;
-        elseif (strcmp (ifmt, "xerr") || index (ifmt, "box"))
+        elseif (strcmp (ifmt, "xerr") || strcmp (ifmt, "box"))
           xdata = varargin{1}(:,i);
           ydata = varargin{2}(:,i);
           xldata = varargin{3}(:,i);
           xudata = xldata;
           ldata = [];
           udata = [];
         else # yerr or boxy
           xdata = varargin{1}(:,i);
           ydata = varargin{2}(:,i);
           ldata = varargin{3}(:,i);
           udata = ldata;
           xldata = [];
           xudata = [];
         endif
       case 4
-        if (strcmp (ifmt, "boxxy") || index (ifmt, "xyerr"))
+        if (strcmp (ifmt, "boxxy") || strcmp (ifmt, "xyerr"))
           xdata = varargin{1}(:,i);
           ydata = varargin{2}(:,i);
           xldata = varargin{3}(:,i);
           xudata = xldata;
           ldata = varargin{4}(:,i);
           udata = ldata;
         elseif (strcmp (ifmt, "xerr") || strcmp (ifmt, "box"))
           xdata = varargin{1}(:,i);
@@ -131,17 +131,17 @@ function h = __errplot__ (fstr, p, varar
           xdata = varargin{1}(:,i);
           ydata = varargin{2}(:,i);
           ldata = varargin{3}(:,i);
           udata = varargin{4}(:,i);
           xldata = [];
           xudata = [];
         endif
       case 6 # boxxy, xyerr
-        if (strcmp (ifmt, "boxxy") || index (ifmt, "xyerr"))
+        if (strcmp (ifmt, "boxxy") || strcmp (ifmt, "xyerr"))
           xdata = varargin{1}(:,i);
           ydata = varargin{2}(:,i);
           xldata = varargin{3}(:,i);
           xudata = varargin{4}(:,i);
           ldata = varargin{5}(:,i);
           udata = varargin{6}(:,i);
         else
           error ("error plot with 6 columns only valid for boxxy and xyerr");
